### DISCUSSION TITLE
What files am I using?

### DISCUSSION THREAD

**Discussion by andresanz** [bootscore/bootscore#454] [Q&A] [answered]
is there a mapping file that tells me what file I'm using for each page/post?  What do I mean? For example, my site is setup for a home page names "home" and a posts page names "posts", but What template file am I using? Footer & Headers - got it! (Header.php & Footer.php, an I copied a version of each to my child theme.

but for my /posts page, what template file am i using?  For archives (2023/04/) tags & categories page, what template file am i using.  I'd like to use teh same for all - if possible.  Lke when am i using page.php vs single.php.

thanks for the help

**Comment by justinkruit** [comment, marked as answer]
If I remember correctly, the plugin Query Monitor should tell you that information.
https://wordpress.org/plugins/query-monitor/

**Reply by andresanz** [reply]
You are awesome. Thank you for the suggestion as it’s just what I needed!

**Comment by kreincke** [comment]
May I add some additional hints:

First of all, WordPress offers a [template hierarchy](https://developer.wordpress.org/themes/basics/template-hierarchy/): In the theme, it expects (at least) the files `index.php`, `page.php`, and `single.php`. If you create a page in/with the WP backend, then WP considers the written text as content for the template `page.php` and tries to present the page content by delivering the file `page.php` to the client browser. The file `index.php` should contain a loop, which gathers all posts, created in the WP backend, and delivers a list of posts to the requesting client browser. If a user selects a post (title, etc) from this list, WP tries to present the content by delivering the file `single.php` to the client machine.

The first main point is, that there does not exist any explicit list of mappings from posts/pages to templates accessible for the normal WP user. Mapping posts and pages to the intended templates is a built-in feature of WordPress.

The second main point is, that *bootScore* delivers a perfect set of templates fulfilling the needs (working style) of WP. As long as you do not want to modify the structural appearance, there is no need to copy any of the templates into your child theme. (It's also a built-in feature of WP, that it looks for those templates in the mother theme missed in the child theme. )

There are two ways, to overrule and/or modify the built-in template method: 

A.) To serve a request to the Site-Url, WP first looks for a file/template ` front-page.php`, then for a file `home.php`, and only then it looks for a file `index.php`.  (For the other template types, WP uses a similar hierarchy.). So, if you do not like *bootScores* default `index.php`, create your own `home.php`  in your child theme. But be careful, if you do not correctly embed the WP loop into your `home.php`, you won't see any list of posts.

B.) What, if you want to use a landing page without any posts, but a separate post list page? For this, WP offers a standard method (which also exists independently from *bootScore*): 
* Via the WP backend, create a page(!) named (for example) `myplp` (my post list page): no content, only the title *myplp*
* Via the WP backend, create a page(!) named (for example) `mylap` (my landing page): fill it with your content.
* Goto `Settings/Readings`, click in the section `Your homepage displays` on `A static page`, and insert into the field `Homepage` *mylap* and into the field `Posts page` *myplp*. [This is the crucial point: now, you have overruled WPs default: `https://yourdomain.xy` will be served with the file `mylap` (whatever it contains) and `https://yourdomain.xy/myplp` with the loop-template (index.php or home.php ...)
* If you want to link your landing page with another template then `page.php`,
  * create any template `xyz.php` - in accordance with the WP working style (but it is also a very good idea, to start with any of the existing *bootScore* templates by copying it into your child theme and modifying it by using the *Bootstrap* methodology). 
  *  goto `Pages/mylap`  and select the name of your template in the section `templates` on the sidebar on the right side. (the name is part of your file `xyz.php`

However, there are a few other minor pitfalls that you should avoid. (For details see [https://fodina.de/streamlining-a/](https://fodina.de/streamlining-a/).)

In hope having helped KR
